tests/sp2n/Test_2as_base.cc

```diff
@@ -0,0 +1,141 @@
+#include <Grid/Grid.h>
+
+#define verbose 0
+
+using namespace Grid;
+
+template<int this_nc>
+static void check_dimensions() {
+    
+    const int this_n = this_nc/2;
+    const int this_algebra_dim = Sp<this_nc>::AlgebraDimension;
+    
+    RealD realA;
+    std::cout << GridLogMessage << "Nc = " << this_n << " 2as dimension is " << Sp_TwoIndex<this_nc, AntiSymmetric>::Dimension << std::endl;
+    std::cout << GridLogMessage << "Nc = " << this_n << " 2s dimension is " << Sp_TwoIndex<this_nc, Symmetric>::Dimension << std::endl;
+    std::cout << GridLogMessage << "Nc = " << this_n << " algebra dimension is " << this_algebra_dim << std::endl;
+    realA = Sp_TwoIndex<this_nc, AntiSymmetric>::Dimension + Sp_TwoIndex<this_nc, Symmetric>::Dimension;
+    std::cout << GridLogMessage << "Checking dim(2AS) + dim(AS) + 1 = Nc * Nc " << this_algebra_dim << std::endl;
+    assert ( realA == this_nc * this_nc - 1); // Nc x Nc = dim(2indxS) + dim(2indxAS) + dim(singlet)
+}
+
+template<int this_nc, TwoIndexSymmetry S>
+static void S_checks() {
+    std::cout << S << std::endl;
+    std::cout << 1 + S * 3 << std::endl;
+}
+    
+    
+template<int this_nc, TwoIndexSymmetry S>
+static void run_base_checks() {
+      std::cout << GridLogMessage << " ****** " << std::endl;
+      std::cout << GridLogMessage << "Running checks for Nc = " << this_nc << " TwoIndex Symmetry = " << S << std::endl;
+      const int this_n = this_nc/2;
+      const int this_irrep_dim = Sp_TwoIndex<this_nc, S>::Dimension;
+      const int this_algebra_dim = Sp<this_nc>::AlgebraDimension;
+      typedef typename Sp_TwoIndex<this_nc, S>::template iGroupMatrix<Complex> Matrix;
+      typedef typename Sp_TwoIndex<this_nc, S>::template iGroupTwoIndexMatrix<Complex> ASMatrix;
```

Comment:
I think this name is misleading because it suggests that this is an anti-symmetric matrix while it is actually two index.